src/coprocessor/dag/dag.rs

```diff
@@ -55,21 +63,47 @@ impl DAGContext {
             req_ctx.isolation_level,
             req_ctx.fill_cache,
         );
+        let cache_key = format!(
+            "{:?} {:?} {:?} {:?}",
+            ranges,
+            req.get_flags(),
+            req.get_output_offsets(),
+            req.get_executors()
+        );
 
         let dag_executor = build_exec(req.take_executors().into_vec(), store, ranges, eval_ctx)?;
         Ok(DAGContext {
             columns: dag_executor.columns,
             has_aggr: dag_executor.has_aggr,
+            has_topn: dag_executor.has_topn,
             req_ctx: req_ctx,
             exec: dag_executor.exec,
             output_offsets: req.take_output_offsets(),
             batch_row_limit: batch_row_limit,
+            cache_key: cache_key,
+            enable_distsql_cache: enable_distsql_cache,
+            distsql_cache_entry_max_size: distsql_cache_entry_max_size,
         })
     }
 
-    pub fn handle_request(&mut self) -> Result<Response> {
+    pub fn handle_request(&mut self, region_id: u64) -> Result<Response> {
         let mut record_cnt = 0;
         let mut chunks = Vec::new();
+        let mut version: u64 = 0;
+        if self.can_cache() {
+            let mut cache = DISTSQL_CACHE.lock();
+            let (rver, entry) =
+                cache.get_region_version_and_cache_entry(region_id, &self.cache_key);
+            version = rver;
+            if let Some(data) = entry {
```

Comment:
If the `start_ts` in the request is small, which means need to read a history version, what would it return?

Reply:
OK, I think we should track start_ts in DistSQLCache.